provisioning_client/tests/prov_transport_amqp_common_ut/prov_transport_amqp_common_ut.c

```diff
@@ -456,6 +456,63 @@ static void on_umock_c_error(UMOCK_C_ERROR_CODE error_code)
 
 static TEST_MUTEX_HANDLE g_testByTest;
 
+char* umock_stringify_BINARY_DATA(const BINARY_DATA* value)
+{
+    char* result = (char*)my_gballoc_malloc(1);
+    (void)value;
+    result[0] = '\0';
+    return result;
+}
+
+int umock_are_equal_BINARY_DATA(const BINARY_DATA* left, const BINARY_DATA* right)
+{
+    int result;
+
+    if (left->length != right->length)
+    {
+        result = 0;
+    }
+    else
+    {
+        if (memcmp(left->bytes, right->bytes, left->length) == 0)
+        {
+            result = 1;
+        }
+        else
+        {
+            result = 0;
+        }
+    }
+
+    return result;
+}
+
+int umock_copy_BINARY_DATA(BINARY_DATA* destination, const BINARY_DATA* source)
```

Comment:
This one is good! That's what we should have everywhere above.

Reply:
The only reason this one works, is when source has a length, its bytes are not null. This might require some more investigation into how the tests are setup might be the key to the other tests using registered BINARY_DATA